### ISSUE TITLE
[enhancement] capture airodump-ng output for handshake check

### ISSUE THREAD

**Issue by IwraStudios** [v1s1t0r1sh3r3/airgeddon#136]
## Description
Capture airodump-ng output (FD 2) to find out if a handshake has been found

## Implementation
Untested code for this:
```bash
 xterm -e airodump-ng -c ${channel} -d ${bssid} -w ${tmpdir}handshake ${interface} 2> tmpcheck.txt 2>&1 &

....

+ checkforhandshake
- ask_et_handshake_file
- if [[ ${yesno} = "y" ]];then
```

Sources:

https://stackoverflow.com/questions/17776383/reading-realtime-output-from-airodump-ng

https://stackoverflow.com/questions/876239/how-can-i-redirect-and-append-both-stdout-and-stderr-to-a-file-with-bash

**Comment by OscarAkaElvis** [comment]
Can you test it and if it's working send a pull request? remember to perform the pull request to airgeddon's dev branch (external PRs on master are not allowed).

Thank you.

**Comment by OscarAkaElvis** [comment]
I think is not as simple as perform a regex on a file... you should launch the regex process in background on an infinite loop searching through the output... and you should control the pid in order to kill it when needed. It could be a good enhancement but is not as simple as it seems at first sight. Waiting anxiously  for your PR. 😸

**Comment by IwraStudios** [comment]
My bad I created it on master, I'll change it

**Comment by v1s1t0r1sh3r3** [comment]
I created the "handshake_automation" branch cloning it from dev. You can start the development there. Thank you.

**Comment by IwraStudios** [comment]
@v1s1t0r1sh3r3 I think being able to set a timeout is a good solution. The timeouts really differ per device/network/signal strength. 

I will see if I can add if for evil twin as well, that shouldn't be a problem. 

Side Note: did you keep in mind when creating the regex that the airodump-ng doesn't store everything on smaller screens, this is one of the major problems I ran into in my initial tests

**Comment by v1s1t0r1sh3r3** [comment]
wow... I tried it and you are right... if the screen is so small and the output is not shown at screen, it's not written in the file.... hmmm.... I have a solution for that, we'll change the system. Keep work on the same direction and I'll change it later, no problem. I know how... but I'm pretty busy this days... probably I'll do it the next week.

**Comment by v1s1t0r1sh3r3** [comment]
Oh my god... finally I got involved again... I pushed a new change with the new method for handshake detection. In fact, I re-used an old function with a little refactor. Now it doesn't matter anymore the size of the screen because now we are not using stdout. Now we are checking using aircrack if the handshake is present in the capture (.cap) file which is a much better way to do it. You can check the changes... now implement it in the same way on the other handshake capture and next week I'll do the timeouts stuff.

Another night sleeping few hours... 🙄

**Comment by IwraStudios** [comment]
Sorry for the trouble, that's a better way to do it though.
I'll test and review the changes as soon as I can.

**Comment by OscarAkaElvis** [comment]
Ok, I milestoned it for next 8.10 version. You didn't accepted the invitation to collaborate on the project. Remember to accept it and to work directly on the handshake_automation branch

**Comment by IwraStudios** [comment]
Alright got the invite now, took some time to get in my inbox

**Comment by v1s1t0r1sh3r3** [comment]
Hey, the shellcheck stuff discussed on some commit comments are now fixed for future versions: https://github.com/koalaman/shellcheck/issues/1235

It seems shellcheck version 0.4.7 has a problem. We reported it and more people reported it. To use shellcheck again we should upgrade or downgrade it.

On the other hand, I commited the changes to have a timeout option on this. It works very well. Test it if you want. And if no more changes are needed I'll close this issue very soon. The changes will be available at next 8.10 version.
